src/crimson/common/errorator.h

```diff
@@ -835,23 +835,44 @@ struct errorator {
   class futurize {
     using vanilla_futurize = seastar::futurize<T>;
 
-    template <class...>
-    struct tuple2future {};
-    template <class... Args>
-    struct tuple2future <std::tuple<Args...>> {
-      using type = future<Args...>;
+    template <class Stored, int Dummy = 0>
+    struct stored_to_future {
+      using type = future<Stored>;
+    };
+    template <int Dummy>
+    struct stored_to_future <seastar::internal::monostate, Dummy> {
+      using type = future<>;
     };
 
   public:
     using type =
-      typename tuple2future<typename vanilla_futurize::value_type>::type;
+      typename stored_to_future<typename vanilla_futurize::value_type>::type;
 
     template <class Func, class... Args>
     static type apply(Func&& func, std::tuple<Args...>&& args) {
       return vanilla_futurize::apply(std::forward<Func>(func),
                                      std::forward<std::tuple<Args...>>(args));
     }
 
+    template <class Func, class... Args>
+    static type invoke(Func&& func, Args&&... args) {
```

Comment:
Any chance to drop the `std::tuple`-taking `apply()` above?

Reply:
to drop the `static type apply(Func&& func, std::tuple<Args...>&& args)` method? i wanted to play safe. will try.

Reply:
@rzarzynski  yeah. seems `apply()` is not used anymore. repushed.

Reply:
:+1: